Add support for state_queryStorageAt RPC

vgeddes opened the pull request:
This is meant to replace the deprecated `state_queryStorage` API, which is marked unsafe and isn't actually callable on most production nodes.

As such `state_queryStorageAt` is the only way to perform a low-level storage query right now.

vgeddes commented:
@cdamian How can I go about regenerating `types/test/test-gen/client_mocks_test.go` to include mocks for the new APIs? The file seems auto-generated.

cdamian commented:
> @cdamian How can I go about regenerating `types/test/test-gen/client_mocks_test.go` to include mocks for the new APIs? The file seems auto-generated.

Hey, you can use - https://github.com/vektra/mockery

vgeddes commented:
Thanks for the pointer! Very useful to know about that.

I've made some additional changes related to mocks. I've generated mocks for all RPCs under `rpc/`, and updated the tests at `types/test/test_gen/client_test.go` to use them. The mocks can be regenerated using `go generate`. Added instructions in the README for how to do that.

This should make it more straightforward for contributors to know that mocks exist and should be updated when adding new APIs.

vgeddes commented:
I'm also assuming that the new mocks can replace these legacy mocks at https://github.com/centrifuge/go-substrate-rpc-client/blob/master/rpc/state/state_test.go at some point?

cdamian commented:
> Thanks for the pointer! Very useful to know about that.
> 
> I've made some additional changes related to mocks. I've generated mocks for all RPCs under `rpc/`, and updated the tests at `types/test/test_gen/client_test.go` to use them. The mocks can be regenerated using `go generate`. Added instructions in the README for how to do that.
> 
> This should make it more straightforward for contributors to know that mocks exist and should be updated when adding new APIs.

@vgeddes awesome stuff!

I would only add a small improvement - creating a make target for generating these, something like:

```
generate-mocks:
	@docker run -v `pwd`:/app -w /app --entrypoint /bin/sh vektra/mockery:v2.13.0-beta.1 -c 'go generate ./...'
```
